base.py

```diff
@@ -219,11 +219,20 @@ def get_columns(self, connection, table_name, schema=None, **kw):
         rows = self._get_table_columns(connection, table_name, schema)
         result = []
         for r in rows:
-            # Take out the more detailed type information
-            # e.g. 'map<int,int>' -> 'map'
-            #      'decimal(10,1)' -> decimal
             col_name = r.name
-            col_type = re.search(r'^\w+', r.type).group(0)
+            col_type = ""
+            if r.type.startswith("AggregateFunction"):
+                # Extract type information from a column
+                # using AggregateFunction
+                # the type from clickhouse will be 
+                # AggregateFunction(sum, Int64) for an Int64 type
+                # remove first 24 chars and remove the last one to get Int64
+                col_type = r.type[23:-1]
```

Comment:
I think you'll need a translation table here. The function state is not always going to be `UInt64`, it can be a bitvector for `uniq()` or really anything. You should be using `sumMerge()` functions etc. for the AggregatingMergeTree tables, so it's probably best to return VARCHAR to any `AggregateFunction` as it's not really interpretable.

Reply:
We use this together with superset so this patch was to first make it possible to even create a table in superset for a clickhouse table that had aggregated function state columns 

I was just going to ask in the superset list what the best way would be to generate metrics that automatically use sumMerge or sum depending on the columns real type
Potentially clickhouse sqla would need to override the default sqla types so it can also give extra type information such as being a 'state' type and superset would then use that for sumMerge, countMerge.

FYI the superset code is here
https://github.com/apache/incubator-superset/blob/master/superset/connectors/sqla/models.py